tools/system_libs.py

```diff
@@ -1105,7 +1105,7 @@ def vary_on(cls):
   def get_default_variation(cls, **kwargs):
     return super(libmalloc, cls).get_default_variation(
       malloc=shared.Settings.MALLOC,
-      is_debug=shared.Settings.DEBUG_LEVEL >= 3,
+      is_debug=False,  # debug mode must be explicitly opted in
```

Comment:
In that case can we just remove this config completely?  I don't like maintaining code that can only be activated by modifying the code itself.

Reply:
Fair enough. How about enabling this on `ASSERTIONS=2`? That would keep it around, and it may be useful in some cases that I'm not thinking of right now. Also we have a test for this I see (that failed when I made this change), so would be nice to keep that working. I pushed that now.

Reply:
Works for me

Reply:
Fair enough. How about enabling this on `ASSERTIONS=2`? That would keep it around, and it may be useful in some cases that I'm not thinking of right now.